internal/controller/components/kueue/kueue_support.go

```diff
@@ -54,3 +70,132 @@ func kueueConfigExtraManifestsPath() odhtypes.ManifestInfo {
 		SourcePath: "ocp-4.17-addons",
 	}
 }
+
+func getManagedNamespaces(ctx context.Context, c client.Client) ([]corev1.Namespace, error) {
+	// Deduped namespaces, since some might have both labels.
+	var uniqueNamespaces = make(map[string]corev1.Namespace)
+
+	// Add all namespaces with management label.
+	if err := collectNamespacesWithPagination(ctx, c, uniqueNamespaces, client.MatchingLabels{
+		KueueManagedAnnotationKey: "true",
+	}); err != nil {
+		return nil, err
+	}
+	// Add namespaces with legacy management label.
+	if err := collectNamespacesWithPagination(ctx, c, uniqueNamespaces, client.MatchingLabels{
+		KueueLegacyManagedAnnotationKey: "true",
+	}); err != nil {
+		return nil, err
+	}
+
+	managedNamespacesList := []corev1.Namespace{}
+	for v := range maps.Values(uniqueNamespaces) {
+		managedNamespacesList = append(managedNamespacesList, v)
+	}
+
+	return managedNamespacesList, nil
+}
+
+func collectNamespacesWithPagination(ctx context.Context, c client.Client, namespaceSet map[string]corev1.Namespace, opts ...client.ListOption) error {
+	lo := client.ListOptions{
+		Limit: NSListLimit,
+	}
+	opts = append(opts, &lo)
+	for {
+		// Listing namespaces with management label
+		namespaces := &corev1.NamespaceList{}
+		if err := c.List(ctx, namespaces, opts...); err != nil {
+			return fmt.Errorf("failed to list namespaces with label %s: %w", KueueManagedAnnotationKey+"=true", err)
+		}
+
+		for _, ns := range namespaces.Items {
+			namespaceSet[ns.Name] = ns
+		}
+
+		if namespaces.Continue == "" {
+			break
+		}
+
+		lo.Continue = namespaces.Continue
+	}
+	return nil
+}
+
+func createDefaultClusterQueue(name string) *unstructured.Unstructured {
+	clusterQueue := &unstructured.Unstructured{}
+	clusterQueue.SetGroupVersionKind(gvk.ClusterQueue)
+	clusterQueue.SetName(name)
```

Comment:
maybe not needed, since the same values are set also to the` clusterQueue.Object `